app/main.py

```diff
@@ -1 +1,57 @@
-#  write your code here
+class Animal:
+    def __init__(self, name: str, appetite: int, is_hungry=True) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            self.is_hungry = False
+            print(f"Eating {self.appetite} food points...")
+            return self.appetite
+        else:
+            return 0
+
+
+class Cat(Animal):
+    def __init__(self, name: str, is_hungry=True) -> None:
+        super().__init__(name, 3, is_hungry)
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+    def __init__(self, name: str, is_hungry=True) -> None:
+        super().__init__(name, 7, is_hungry)
+
+    @staticmethod
+    def bring_slippers():
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list[Animal]) -> int:
+    food_total_points = 0
+    for animal in animals:
+        food_total_points += animal.feed()
+    return food_total_points
+
+
+lion = Animal("Lion", 25)
+print(lion.print_name())  # "Hello, I'm Lion"
```

Comment:
The `print_name()` method already prints the name, so using `print(lion.print_name())` will print `None` after the name. You might want to call `lion.print_name()` without wrapping it in `print()`.